patches/0004-Use-crypto-backends.patch

```diff
@@ -379,19 +380,164 @@ index 33942467784ad3..0282ffa9fa23c8 100644
+ 	return &gcmFallback{cipher: cipher, nonceSize: nonceSize, tagSize: tagSize}, nil
+ }
+ 
++// customGCMWrapper wraps an AEAD implementation to expose only the AEAD interface methods.
++// This prevents extra exported methods from causing test failures in cryptotest.NoExtraMethods.
++type customGCMWrapper struct {
++	aead AEAD
++}
++
++func (g customGCMWrapper) NonceSize() int { return g.aead.NonceSize() }
++
++func (g customGCMWrapper) Overhead() int { return g.aead.Overhead() }
++
++func (g customGCMWrapper) Seal(dst, nonce, plaintext, additionalData []byte) []byte {
++	return g.aead.Seal(dst, nonce, plaintext, additionalData)
++}
++
++func (g customGCMWrapper) Open(dst, nonce, ciphertext, additionalData []byte) ([]byte, error) {
++	return g.aead.Open(dst, nonce, ciphertext, additionalData)
++}
++
+ // gcmFallback is only used for non-AES ciphers, which regrettably we
+ // theoretically support. It's a copy of the generic implementation from
+ // crypto/internal/fips140/aes/gcm/gcm_generic.go, refer to that file for more details.
 diff --git a/src/crypto/cipher/gcm_test.go b/src/crypto/cipher/gcm_test.go
-index e574822c9aa6f8..cf121d9e5f654d 100644
+index e574822c9aa6f8..f32707146a369a 100644
 --- a/src/crypto/cipher/gcm_test.go
 +++ b/src/crypto/cipher/gcm_test.go
-@@ -8,7 +8,7 @@ import (
+@@ -8,7 +8,6 @@ import (
  	"bytes"
  	"crypto/aes"
  	"crypto/cipher"
 -	"crypto/internal/boring"
-+	boring "crypto/internal/backend"
  	"crypto/internal/cryptotest"
  	"crypto/internal/fips140"
  	fipsaes "crypto/internal/fips140/aes"
+@@ -17,6 +16,7 @@ import (
+ 	"encoding/hex"
+ 	"errors"
+ 	"fmt"
++	"internal/goexperiment"
+ 	"io"
+ 	"reflect"
+ 	"testing"
+@@ -727,9 +727,13 @@ func testGCMAEAD(t *testing.T, newCipher func(key []byte) cipher.Block) {
+ 
+ 			// Test NewGCMWithRandomNonce.
+ 			t.Run("GCMWithRandomNonce", func(t *testing.T) {
+-				if _, ok := block.(*wrapper); ok || boring.Enabled {
++				if _, ok := block.(*wrapper); ok || goexperiment.BoringCrypto {
+ 					t.Skip("NewGCMWithRandomNonce requires an AES block cipher")
+ 				}
++				a, _ := cipher.NewGCMWithRandomNonce(block)
++				if a.NonceSize() != 0 {
++					t.Errorf("NewGCMWithRandomNonce returned AEAD with non-zero nonce size: %d", a.NonceSize())
++				}
+ 				cryptotest.TestAEAD(t, func() (cipher.AEAD, error) { return cipher.NewGCMWithRandomNonce(block) })
+ 			})
+ 		})
+@@ -752,10 +756,13 @@ func TestGCMExtraMethods(t *testing.T) {
+ 		})
+ 		t.Run("NewGCMWithRandomNonce", func(t *testing.T) {
+ 			block := newCipher(make([]byte, 16))
+-			if _, ok := block.(*wrapper); ok || boring.Enabled {
++			if _, ok := block.(*wrapper); ok || goexperiment.BoringCrypto {
+ 				t.Skip("NewGCMWithRandomNonce requires an AES block cipher")
+ 			}
+ 			a, _ := cipher.NewGCMWithRandomNonce(block)
++			if a.NonceSize() != 0 {
++				t.Errorf("NewGCMWithRandomNonce returned AEAD with non-zero nonce size: %d", a.NonceSize())
++			}
```

Comment:
Why is this `a.NonceSize() != 0` test happening in `NewGCMWithRandomNonce` too? `GCMWithRandomNonce` covers more key sizes, and I'm not spotting other differences.

On the other hand, it seems odd that `GCMWithRandomNonce` has the *real* coverage of `cipher.NewGCMWithRandomNonce`, rather than `NewGCMWithRandomNonce`.

For the sake of diff minimization, I think removing this makes sense even though the test name doesn't line up. However, for upstreaming, think might need to be worked out better.